### DISCUSSION TITLE
Using  Netbox as a blind user with screen reading software

### DISCUSSION THREAD

**Discussion by sonor3000** [netbox-community/netbox#13084] [Help Wanted!]
We start to use Netbox in our company and as a systemengineer I also have to use it. Because I am blind I need to work with a screen reading software on my computer, e.g. Jaws or NVDA on Windows, VoiceOver on a Mac or Orca on Linux. I've tried all combinations of screen readers and browsers on a Mac with VoiceOver and on Windows or Linux with Jaws, NVDA or Orca and Firefox and Brave, but in some situations the webinterface is not useable for me.

Reading the stored informations is OK, at least I had no problem so far, but editing items is quite difficult or even inpossible no matter which browser and screen reading software on different operating systems I use :-(.

The biggest problems seem to be the list boxes and search dialogs that are used when I want to change the information e.g. for a virtual machine, an ip address, a network interface and so on. I think this listboxes, or whatever these elements are, are used all over the whole Netbox webinterface, so the problem seems not be specific to one type of item.

Is there something I can do to get a more accessible webinterface in Netbox which is easier to use with a screen reading software? Are there e.g. special themes I could install that are more more screen reader friendly?

I also searched around to work via a CLI connected to our Netbox instance, but all I've found seems not to have all features included I'd need. Or is there a CLI I could use to create, edit and remove e.g. virtual machines and the data related to this item in Netbox?

Currently handling virtual machines, related ip addresses, tags and services would be the most important task for me that I want to handle with Netbox, but other tasks might be come up in future.

Do you have any ideas that might help to get a more useable webinterface or to perform the tasks I need via a CLI, the API or however and is it planed to have a more accessible webinterface for Netbox in future? If I can help to get such a more screen reader friendly webinterface I'd be happy to support where I am able to.
 
Thanks a lot for any helpfull hint or idea!

**Comment by ziggekatten** [comment]
As an cli, the closest are pynetbox. To make it an simple cli client, you would however need to write some python code to wrap things into simple commands.

I have spent quite some work on braille printer output in my days, but never worked with keyboard/screen reader interactions. What would be the guidelines for availability/usability?

**Comment by jsenecal** [comment]
A netbox cli client working in conjunction with netbox API should be fairly straightforward to build given that the API Schema is valid.
This, however is a project on its own unfortunately. Sorry if I am not of more help.

**Comment by martinum4** [comment]
If you currently use mainly Windows, and most screen readers there are optimized for office, i'd suggest just using excel for your task, the NetBox-API works quite well with it, i personally use it to print labels for our racks and check which names we already have assigned.
You could autogenerate deeplinks in it too, which directly take you to delete/edit/whatever…
